client-app/shared/cart/components/add-to-cart.vue

```diff
@@ -63,10 +56,13 @@ interface IProps {
 }
 
 const product = toRef(props, "product");
-const { cart, addToCart, changeItemQuantity } = useShortCart();
+const { cart, addToCart, changeItemQuantity, changeCartConfiguredItem } = useShortCart();
 const { t } = useI18n();
 const ga = useGoogleAnalytics();
 const { translate } = useErrorsTranslator<ValidationErrorType>("validation_error");
+const { lineItemId: configurableLineItemId } = useUrlSearchParams<{ lineItemId: string | undefined }>("history", {
```

Comment:
why useUrlSearchParams it is better then 
![image](https://github.com/user-attachments/assets/a337f923-c4c0-4f3a-9b64-7ff2ed18a994)

Reply:
You are right, using useUrlSearchParams is a bit redundant here. Created utils function and used it instead.